guppylang/std/_internal/compiler/array.py

```diff
@@ -332,7 +370,9 @@ def _build_call_getitem(
 
     def compile_with_inouts(self, args: list[Wire]) -> CallReturnWires:
         [array, idx] = args
-        if self.elem_ty.type_bound() == ht.TypeBound.Any:
+        [elem_ty_arg, _] = self.type_args
+        assert isinstance(elem_ty_arg, TypeArg)
+        if not elem_ty_arg.ty.copyable:
```

Comment:
We need to relax this constraint to non-copyable since we have to avoid `get`s on nested arrays: Those only work for copyable element types